internal/iam/repository_role.go

```diff
@@ -184,16 +187,141 @@ func (r *Repository) DeleteRole(ctx context.Context, withPublicId string, _ ...O
 	return rowsDeleted, nil
 }
 
-// ListRoles lists roles in the given scopes and supports WithLimit option.
-func (r *Repository) ListRoles(ctx context.Context, withScopeIds []string, opt ...Option) ([]*Role, error) {
-	const op = "iam.(Repository).ListRoles"
+// listRoles lists roles in the given scopes and supports WithLimit option.
+func (r *Repository) listRoles(ctx context.Context, withScopeIds []string, opt ...Option) ([]*Role, time.Time, error) {
+	const op = "iam.(Repository).listRoles"
 	if len(withScopeIds) == 0 {
-		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing scope ids")
+		return nil, time.Time{}, errors.New(ctx, errors.InvalidParameter, op, "missing scope id")
 	}
-	var roles []*Role
-	err := r.list(ctx, &roles, "scope_id in (?)", []any{withScopeIds}, opt...)
+	opts := getOpts(opt...)
+
+	limit := r.defaultLimit
+	if opts.withLimit != 0 {
+		// non-zero signals an override of the default limit for the repo.
+		limit = opts.withLimit
+	}
+
+	var args []any
+	whereClause := "scope_id in @scope_ids"
+	args = append(args, sql.Named("scope_ids", withScopeIds))
```

Comment:
Does this work? In my experience I'd had to manually expand slices into multiple arguments when using named parameters.

Reply:
🤷🏻 I guess it must've been something else that broke for me when I did something like this

Reply:
I'm gonna submit a PR to replace the instances of the workaround I implemented for this then :P